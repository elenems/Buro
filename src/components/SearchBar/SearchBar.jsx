import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import IconButton from "@material-ui/core/IconButton";
import SearchIcon from "@material-ui/icons/Search";
import Box from "@material-ui/core/Box";
import Select from "@material-ui/core/Select";
import FormControl from "@material-ui/core/FormControl";
import MenuItem from "@material-ui/core/MenuItem";
import { makeStyles } from "@material-ui/core/styles";
import FormHelperText from "@material-ui/core/FormHelperText";

const useStyles = makeStyles(theme => ({
  root: {
    display: "flex",
    flexWrap: "wrap"
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120
  },
  selectEmpty: {
    marginTop: theme.spacing(2)
  }
}));
export default function SearchBar() {
  const classes = useStyles();
  const [region, setRegion] = useState("Lviv");
  const handleChange = e => {
    const region = e.target.value;
    setRegion(region);
  };
  return (
    <Box
      display="flex"
      justifyContent="center"
      alignItems="baseline"
    >
      <FormControl className={classes.formControl}>
        <Select value={region} name="region" onChange={handleChange} autoWidth>
          <MenuItem value="Lviv">Львів</MenuItem>
          <MenuItem value="Kharkiv">Харків</MenuItem>
          <MenuItem value="Kyiv">Київ</MenuItem>
        </Select>
        <FormHelperText>Регіон</FormHelperText>
      </FormControl>
      <TextField
        fullWidth={true}
        id="search"
        label="Шукати майно"
        type="search"
        margin="normal"
      />

      <IconButton color="primary" aria-label="Search">
        <SearchIcon />
      </IconButton>
    </Box>
  );
}
