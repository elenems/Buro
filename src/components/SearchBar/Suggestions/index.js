import Suggestions from "./Suggestions.jsx";
export default Suggestions;
