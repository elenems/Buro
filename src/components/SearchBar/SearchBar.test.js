import React from "react";
import { shallow, mount } from "enzyme";
import SearchBar from "../SearchBar";
import Select from "@material-ui/core/Select";
import { BrowserRouter } from "react-router-dom";

describe("SearchBar", () => {
  it("Renders", () => {
    const wrapper = shallow(<SearchBar />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Sets region", () => {
    const wrapper = shallow(<SearchBar />);
    wrapper.find(Select).simulate("change", { target: { value: "Kyiv" } });
    expect(wrapper.find(Select).prop("value")).toBe("Kyiv");
  });

  it("Sets title", () => {
    const wrapper = shallow(<SearchBar />);
    wrapper
      .find("#search")
      .simulate("change", { target: { value: "Some value" } });
    expect(wrapper.find("#search").prop("value")).toBe("Some value");
  });

});
