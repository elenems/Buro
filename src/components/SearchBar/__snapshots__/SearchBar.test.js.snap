// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Renders 1`] = `
"<Styled(MuiBox) style={{...}} display=\\"flex\\" justifyContent=\\"center\\" alignItems=\\"baseline\\">
  <WithStyles(ForwardRef(FormControl)) className=\\"makeStyles-formControl-2\\">
    <WithStyles(ForwardRef(Select)) value=\\"Lviv\\" name=\\"region\\" onChange={[Function: handleChange]} autoWidth={true}>
      <WithStyles(ForwardRef(MenuItem)) value=\\"Lviv\\">
        Львів
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem)) value=\\"Kharkiv\\">
        Харків
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem)) value=\\"Kyiv\\">
        Київ
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <WithStyles(ForwardRef(FormHelperText))>
      Регіон
    </WithStyles(ForwardRef(FormHelperText))>
  </WithStyles(ForwardRef(FormControl))>
  <WithStyles(ForwardRef(TextField)) autoComplete=\\"off\\" fullWidth={true} id=\\"search\\" label=\\"Шукати майно\\" type=\\"search\\" margin=\\"normal\\" value=\\"\\" onChange={[Function: handleTitleChange]} />
  <WithStyles(ForwardRef(IconButton)) color=\\"primary\\" aria-label=\\"Search\\">
    <SearchIcon />
  </WithStyles(ForwardRef(IconButton))>
  <withRouter(Suggestions) items={{...}} />
</Styled(MuiBox)>"
`;
